web-wallet/src/lib/components/ApproveMigration/ApproveMigration.svelte

```diff
@@ -0,0 +1,140 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { mdiAlertOutline, mdiTimerSand } from "@mdi/js";
+  import { getAccount, waitForTransactionReceipt } from "@wagmi/core";
+  import { isHex } from "viem";
+  import { createEventDispatcher } from "svelte";
+  import { Button, Icon } from "$lib/dusk/components";
+  import { wagmiConfig } from "$lib/migration/walletConnection";
+  import { allowance, approve } from "$lib/migration/migration";
+  import { createDataStore } from "$lib/dusk/svelte-stores";
+
+  /** @type {number} */
+  export let amount;
+
+  /** @type {`0x${string}`} */
+  export let chainContract;
+
+  /** @type {`0x${string}`} */
+  export let migrationContract;
+
+  /** @type {import("svelte/store").Writable<Boolean>}*/
+  export let hasApprovedCoin;
+
+  const dispatch = createEventDispatcher();
+
+  const approvalTxStore = createDataStore(handleApprove);
+
+  const retrieveAccount = () =>
+    new Promise(() => {
+      return getAccount(wagmiConfig);
+    });
```

Comment:
I think this is doing too much perhaps for nothing? I guess within the `handleApprove` you can just do:

```
const account = getAccount(wagmiConfig);
if (account.address) {
  await checkAllowance(account.address);
}
```

and perhaps throw some error if `account.address` does not exist. 👀